### ISSUE TITLE
Support multiple antiAffinity statements

### ISSUE THREAD

**Issue by bcbrockway** [prometheus-operator/kube-prometheus#1090]
**What is missing?**

We can't currently add affinityTerms using the provided "antiaffinity" function in `addons/` because the function doesn't use the `+:` notation on the keys it creates. The `podAntiAffinityTopologyKey` key in `values` also only supports one topology key.

**Why do we need it?**

We would like to add an antiAffinity rule for zone, as well as hostname so that prometheus, alertmanager, etc. instances are put onto different hosts in different zones (the hostname rule would be a fallback if the zone rule was unable to be carried out).

**Environment**

* kube-prometheus version:

    `dff8d8c11ccace01468ba7c4bae25784e5113b45`

**Anything else we need to know?**:

**Comment by paulfantom** [comment]
I agree it would be good to improve this addon. Do you want to contribute a fix?

**Comment by slashpai** [comment]
Not familiar with jsonnet yet but if you can give some starting points I can look at it :)
